### PR TITLE
Fixes missing method `path_for` when using MirrorService with DiskService as the primary service

### LINKED ISSUE #35252: ActiveStorage raises with missing method `path_for` when using MirrorService with DiskService as the primary service
Upgrading to Rails 5.2.2 (from 5.2.1) breaks ActiveStorage using MirrorService with DiskService as the primary service.

### Steps to reproduce

Use ActiveStorage and setup a mirrored service with Disk as primary, e.g.

```yml
local:
  service: Disk
  root: <%= Rails.root.join("storage") %>

amazon:
  ...

mirror:
  service: Mirror
  primary: local
  mirrors: [ amazon ]
```

When trying to fetch a stored resource it crashes.

### Expected behavior

Expects to be able to use a mirror with a disk service as primary service.

### Actual behavior

When fetching the resources while having the disk service as primary service it crashes.

### System configuration
**Rails version**:
5.2.2

**Ruby version**:
ruby 2.5.0p0

---

It raises on missing method `path_for` on line: https://github.com/rails/rails/blob/5-2-stable/activestorage/app/controllers/active_storage/disk_controller.rb#L12

Changing the primary service to amazon works, as well as using rails 5.2.1 instead of 5.2.2.

As far as I can tell this problem was introduced with this change: https://github.com/rails/rails/commit/62bb663174e6a16775d619fbe17073408c116113

#### Some random thoughts:

A potential solution could be to delegate `:path_for` to the primary service here:
https://github.com/rails/rails/blob/5-2-stable/activestorage/lib/active_storage/service/mirror_service.rb#L12
My problem with this is that the documentation for the MirrorService states that it is a meant to handle all uploads. This would both repurpose it to also handle fetching, but maybe more importantly, also change based on the need of the internal implementation of a specific service which isn't something that the other services are using.

Another option could be to change https://github.com/rails/rails/blob/5-2-stable/activestorage/app/controllers/active_storage/disk_controller.rb#L12 to check if the service is a mirrorservice and then use the primary service to call `path_for`, instead of on the mirror service. Using two different kind of services seems more error prone in the future and moves away from using an common interface for the services.

### TL;DR:
The problem seems to be that the `DiskController` relies on a `path_for` method that is defined in the `DiskService` which isn't exposed/forwarded in the `MirrorService`.

A potential solution would be to extend the "interface" in the `MirrorService` to allow for the `path_for` method that is currently only used by the DiskService. That is, add `path_for` to `delegate ..., to: :primary` in the MirrorService

### PR DESCRIPTION
**Pull request by abhaynikam**
fixes #35252 

@r? @georgeclaghorn 

cc/ @marcusmalmberg

### CODE REVIEW

**Comment by georgeclaghorn** [comment]
Thanks, @abhaynikam. This needs a test.

**Comment by abhaynikam** [comment]
@georgeclaghorn : Added test cases. Please let me know if we should add more cases.

File: activestorage/test/service/mirror_service_test.rb [hunk 1]
```diff
@@ -61,4 +61,10 @@ class ActiveStorage::Service::MirrorServiceTest < ActiveSupport::TestCase
         @service.url(@key, expires_in: 2.minutes, disposition: :inline, filename: filename, content_type: "text/plain")
     end
   end
+
+  test "path for file in all mirror services" do
+    SERVICE.mirrors.each do |mirror|
+      assert mirror.path_for(@key)
+    end
+  end
```
Comment (georgeclaghorn) [inline activestorage/test/service/mirror_service_test.rb]: This test passes without the code change. We want to test that the mirror delegates `path_for` to the primary:

```ruby
test "path for file in primary service" do
  assert_equal @service.primary.path_for(@key), @service.url_for(@key)
end
```

File: activestorage/test/service/mirror_service_test.rb [hunk 2]
```diff
@@ -61,4 +61,8 @@ class ActiveStorage::Service::MirrorServiceTest < ActiveSupport::TestCase
         @service.url(@key, expires_in: 2.minutes, disposition: :inline, filename: filename, content_type: "text/plain")
     end
   end
+
+  test "path for file in primary service" do
+    assert_equal @service.primary.path_for(@key), @service.url_for(@key)
```
Comment (georgeclaghorn) [inline activestorage/test/service/mirror_service_test.rb]: Sorry, I used `url_for` in my example when I meant `path_for`. 🤦‍♂️

```suggestion
    assert_equal @service.primary.path_for(@key), @service.path_for(@key)
```

Comment (abhaynikam) [reply activestorage/test/service/mirror_service_test.rb]: Sorry. My bad. I should have checked it.

**Comment by abhaynikam** [comment]
@georgeclaghorn : Done with the changes.

**Comment by georgeclaghorn** [comment]
Thank you!

**Pull request merged**